File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    const setLen = [...new Set(nums)].length;
    const max = nums.length / 2;
    
    return setLen < max ? setLen : max;
    
    //return [...new Set(nums)].length < nums.length/2 ? [...new Set(nums)].length : nums.length/2;
}

File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    n = len(nums) / 2
    k = 0
    arr = []
    while(n != 0 and k < len(nums)):
        if nums[k] not in arr:
            arr.append(nums[k])
            n = n - 1
            answer = answer + 1
        k = k + 1
        
    return answer